File: Rust/Demo/src/station.rs
pub struct Station {
    distance: Option<f64>,
    latitude: Option<f64>,
    longitude: Option<f64>,
    use_count: Option<i32>,
    id: Option<String>,
    name: Option<String>,
    quality: Option<i32>,
    contribution: Option<f64>,
}

impl Station {
    // Constructor
    pub fn new() -> Station {
        Station {
            distance: None,
            latitude: None,
            longitude: None,
            use_count: None,
            id: None,
            name: None,
            quality: None,
            contribution: None,
        }
    }

    // Getters and setters
    pub fn distance(&self) -> Option<f64> {
        self.distance
    }

    pub fn set_distance(&mut self, distance: Option<f64>) {
        self.distance = distance;
    }

    pub fn latitude(&self) -> Option<f64> {
        self.latitude
    }

    pub fn set_latitude(&mut self, latitude: Option<f64>) {
        self.latitude = latitude;
    }

    pub fn longitude(&self) -> Option<f64> {
        self.longitude
    }

    pub fn set_longitude(&mut self, longitude: Option<f64>) {
        self.longitude = longitude;
    }

    pub fn use_count(&self) -> Option<i32> {
        self.use_count
    }

    pub fn set_use_count(&mut self, use_count: Option<i32>) {
        self.use_count = use_count;
    }

    pub fn id(&self) -> &Option<String> {
        &self.id
    }

    pub fn set_id(&mut self, id: Option<String>) {
        self.id = id;
    }

    pub fn name(&self) -> &Option<String> {
        &self.name
    }

    pub fn set_name(&mut self, name: Option<String>) {
        self.name = name;
    }

    pub fn quality(&self) -> Option<i32> {
        self.quality
    }

    pub fn set_quality(&mut self, quality: Option<i32>) {
        self.quality = quality;
    }

    pub fn contribution(&self) -> Option<f64> {
        self.contribution
    }

    pub fn set_contribution(&mut self, contribution: Option<f64>) {
        self.contribution = contribution;
    }
}

File: Scala/Demo/src/main/scala/Station.scala
class Station(
               private var _distance: Option[Double] = None,
               private var _latitude: Option[Double] = None,
               private var _longitude: Option[Double] = None,
               private var _useCount: Option[Int] = None,
               private var _id: Option[String] = None,
               private var _name: Option[String] = None,
               private var _quality: Option[Int] = None,
               private var _contribution: Option[Double] = None
             ) {

  def this() = this(None, None, None, None, None, None, None, None)

  def distance: Option[Double] = _distance
  def distance_=(distance: Option[Double]): Unit = {
    _distance = distance
  }

  def latitude: Option[Double] = _latitude
  def latitude_=(latitude: Option[Double]): Unit = {
    _latitude = latitude
  }

  def longitude: Option[Double] = _longitude
  def longitude_=(longitude: Option[Double]): Unit = {
    _longitude = longitude
  }

  def useCount: Option[Int] = _useCount
  def useCount_=(useCount: Option[Int]): Unit = {
    _useCount = useCount
  }

  def id: Option[String] = _id
  def id_=(id: Option[String]): Unit = {
    _id = id
  }

  def name: Option[String] = _name
  def name_=(name: Option[String]): Unit = {
    _name = name
  }

  def quality: Option[Int] = _quality
  def quality_=(quality: Option[Int]): Unit = {
    _quality = quality
  }

  def contribution: Option[Double] = _contribution
  def contribution_=(contribution: Option[Double]): Unit = {
    _contribution = contribution
  }
}
